### ISSUE TITLE
`meta-json` variable can't be used in a HTML template when some metadata contains HTML tags

### ISSUE THREAD

**Issue by cderv** [jgm/pandoc#8048]
For the context, I am using `meta-json` variable to pass Pandoc metadata to HTML browser so that I can access the values in some JS processing. For this I am inserting this in my HTML template

````html
<script id="pandoc-meta" type="application/json">$meta-json$</script>
````

which allow me to retrieve the metadata values in JS using
````js
let el = document.getElementById('pandoc-meta');
pandocMeta = el ? JSON.parse(el.firstChild.data) : {};
````

This was working well until some document are setting `header-includes` for example. `header-includes` meta will contains usually some HTML code using HTML node tags. And this is does not play well to have HTML as it messes the parsing in browser. 

Here is an example. 

* `test.md`
````markdown
---
header-includes: |
    <script>console.log("test")</script>
---
````

* Here is a small template `template.html` for the purpose of seing what is included:
````html
<html>
    <head>
        <script id="pandoc-meta" type="application/json">
            $meta-json$
        </script>
    </head>
    <body>
        <pre><code>$meta-json$</code></pre>
    </body>
</html>
````

Let's render with 
````sh
pandoc -t html test.md -o test.html --template template.html
````

The HTML produced is 
````html
<html>
    <head>
        <script id="pandoc-meta" type="application/json">
            {"header-includes":"<script>console.log(\"test\")</script>"}
        </script>
    </head>
    <body>
        <pre><code>{"header-includes":"<script>console.log(\"test\")</script>"}</code></pre>
    </body>
</html>
````
`$meta-json$` was replaced by metadatas in JSON representation, but the HTML content is not escaped or encoded. 

If  you open the `test.html` document, you'll notice some problem in content parsed as the `<script>` in the JSON is seen as part of the HTML structure. 

![image](https://user-images.githubusercontent.com/6791940/165779885-bb06fb14-442d-4c6e-955c-757c3cef024c.png)

This leads me to a question: Is `$meta-json$` not intended to be inserted into a HTML template ? 
It seemed like a good solution to pass some data to external tools processing the resulting HTML. My understanding of https://github.com/jgm/pandoc/commit/4361dc0245a65d4f24f2df062684cdb1a0c3bc5a is that it was one of the purpose.

Is there any way to select which meta to include in the JSON string ? Or any other to achieve something similar to export metadata ? 

I don't know what it would imply to encode the resulting JSON string in `meta-json` for the output format inclusion. 

Among solution, I found that escaping the closing `</script>` like `<\/script>` for inclusion in `<script id="pandoc-meta" type="application/json">` seems to work (which is described in https://www.w3.org/TR/html4/appendix/notes.html#h-B.3.2 and still works). HTML escaping is another option at least using `&lt;` for `<`, which works for inclusion in body to show the metadata, 

I did not found solution to process `$meta-json$` to that purpose. Having a pipe function for template to do string replacement or string escaping would help.

Overall, is this something that could be improve ? Or is `$meta-json$` just mean to be used to create a json data file from a template ? with only the variable ?

Thanks.

**Comment by jgm** [comment]
Correct, `meta-json` is treated as a variable value and inserted directly with no escaping.
 In your case you can work around this, I think, with
```html
<script type="application/json">//<![CDATA[
   $meta-json$
//]]></script>
```

**Comment by jgm** [comment]
I agree that it might be desirable to escape it for the target format, but it's a bit tricky to anticipate what is needed for the full range of cases.  e.g. there might be some formats in which you'd want to put meta-json in a special context where it *shouldn't* be escaped.

**Comment by RLesur** [comment]
> I agree that it might be desirable to escape it for the target format, but it's a bit tricky to anticipate what is needed for the full range of cases. e.g. there might be some formats in which you'd want to put meta-json in a special context where it _shouldn't_ be escaped.

I wonder whether an escaping pipe would make sense (to be used in an HTML template, e.g. `meta-json/HTMLescape`)
